packages/babel-types/scripts/generators/typescript.js

```diff
@@ -75,18 +76,20 @@ for (const type in t.NODE_FIELDS) {
       typeAnnotation += " | null";
     }
 
-    if (areAllRemainingFieldsNullable(fieldName, fieldNames, fields)) {
-      args.push(
-        `${t.toBindingIdentifierName(fieldName)}${
-          isNullable(field) ? "?:" : ":"
-        } ${typeAnnotation}`
-      );
-    } else {
-      args.push(
-        `${t.toBindingIdentifierName(fieldName)}: ${typeAnnotation}${
-          isNullable(field) ? " | undefined" : ""
-        }`
-      );
+    if (builderNames.includes(fieldName)) {
```

Comment:
Wouldn't it be better to directly iterate over `builderNames` instead of `fieldNames`? Since `fieldNames` is generated from an object, I think that we risk changing the order of its keys without thinking about this problem.

Reply:
The `fieldNames` has been sorted according to the orders of `buildKeys` and anything not in the `builders` are at tails.
https://github.com/babel/babel/blob/d0a8982c124ac495b6008f9e5958ec8ad9b5a3f8/packages/babel-types/scripts/generators/typescript.js#L349